src/components/Expander/ExpanderGroup.tsx

```diff
@@ -82,43 +99,72 @@ const OpenAllButton = ({
   onClick,
   OpenAll,
   CloseAll,
+  onClickAll,
 }: OpenAllButtonProps) => {
   return !!allOpen
-    ? openAllButtonProps(onClick)(CloseAll)
-    : openAllButtonProps(onClick)(OpenAll);
+    ? openAllButtonProps(onClick, onClickAll)(CloseAll)
+    : openAllButtonProps(onClick, onClickAll)(OpenAll);
+};
+
+const InitialStateOfExpanders = (
+  children: Array<React.ReactElement<ExpanderProps>>,
+) => {
+  const openExpanders: number[] = [];
+  React.Children.forEach(children, (child, index) => {
+    if (React.isValidElement(child)) {
+      if (child.props.defaultOpen || child.props.open) {
+        openExpanders.push(index);
+      }
+    }
+  });
+  return openExpanders;
 };
 
 export class ExpanderGroup extends Component<ExpanderGroupProps> {
   state: ExpanderGroupState = {
-    openExpanders: [],
+    openExpanders: InitialStateOfExpanders(this.props.children),
+    toggleAllExpanderState: {
+      toState:
+        InitialStateOfExpanders(this.props.children).length ===
+          this.props.children.length && this.props.children.length > 0,
+    },
   };
 
   handleClick = (index: number = 0) => {
```

Comment:
I'd rename this since we don't handle clicks with this anymore?

Reply:
This is now renamed to be `handleExpanderOpenChange`